plotting/quicklook_generator.py

```diff
@@ -405,12 +405,153 @@ def ultra_general_rates_2_plot(self) -> None:
+        if self.data_set is None:
+            raise ValueError("Must load in a dataset.")
+
+        # Time data
+        epoch = self.data_set["epoch"].values
+        epoch_dt = convert_j2000_to_utc(epoch)
+
+        # Energy values from strings to numbers
+        energy_labels = self.data_set["energy_label"].values.astype(float)
+
+        # Get count rate data
+        swp_rates = self.data_set["swp_pcem_rate"].values
+        pui_rates = self.data_set["swp_scem_rate"].values
+
+        fig, axes = plt.subplots(2, 1, figsize=(12, 8), sharex=True, sharey=True)
+
+        # Plot SW
+        pcm1 = axes[0].pcolormesh(
+            epoch_dt, energy_labels, swp_rates.T, shading="auto", cmap="viridis"
+        )
+        axes[0].set_title("SWP PCEM Count Rate")
+        axes[0].set_ylabel("Energy per charge (eV/q)")
+        fig.colorbar(pcm1, ax=axes[0], label="Counts")
+
+        # Plot PUI
+        pcm2 = axes[1].pcolormesh(
+            epoch_dt, energy_labels, pui_rates.T, shading="auto", cmap="viridis"
+        )
+        axes[1].set_title("SWP SCEM Count Rate")
+        axes[1].set_xlabel("Time (UTC)")
+        axes[1].set_ylabel("Energy per charge (eV/q)")
+        fig.colorbar(pcm2, ax=axes[1], label="Counts")
+
+        plt.tight_layout()
+        plt.show()
+
+    def swapi_absolute_detection_efficiency(self) -> None:
+        """
+        Graph SWAPI absolute detection efficiency.
+
+        Notes:
+        --------
+            Calculate using coincidence^2 / (primary*secondary).
+        """
+        if self.data_set is None:
+            raise ValueError("Must load in a dataset.")
+
+        # Time data
+        epoch = self.data_set["epoch"].values
+        epoch_dt = convert_j2000_to_utc(epoch)
+
+        # Get count rate data
+        pcem_rate = self.data_set["swp_pcem_rate"]
+        pcem_rate_sum = pcem_rate.sum(dim="energy")
+
+        scem_rate = self.data_set["swp_scem_rate"]
+        scem_rate_sum = scem_rate.sum(dim="energy")
+
+        coin_rate = self.data_set["swp_coin_rate"]
+        coin_rate_sum = coin_rate.sum(dim="energy")
+
+        denom = pcem_rate_sum * scem_rate_sum
+        detection_efficiency = (coin_rate_sum**2) / (denom)
+        detection_efficiency = detection_efficiency.where(denom != 0)
+
+        plt.plot(epoch_dt, detection_efficiency)
+        plt.title("SWAPI Absolute Detection Efficiency")
+        plt.xlabel("Time (UTC)")
+        plt.ylabel("Coincidence² / (Primary × Secondary)")
+        plt.grid(True)
+        plt.tight_layout()
+        plt.show()
+
+    def swapi_count_line(self) -> None:
+        """Generate SWAPI count rates line plot."""
+
+        if self.data_set is None:
+            raise ValueError("Must load in a dataset.")
+
+        # Time data
+        epoch = self.data_set["epoch"].values
+        epoch_dt = convert_j2000_to_utc(epoch)
+
+        # Get count data
+        # TODO: Ensure energy=0 is correct
+        pcem_counts = self.data_set["swp_pcem_rate"]
```

Comment:
NOTE that counts are from L1 file and rates are from L2 file.
```suggestion
        pcem_counts = self.data_set["swp_pcem_counts"]
```

Reply:
Unfortunately, there isn't a variable in the cdf named "swp_pcem_counts". I am assuming that the rate is the wrong variable?

Reply:
That variable is in a l1 CDF file. rates are in L2 CDF file. Don't hesitate to setup a zoom if you like to go through my comments.

Reply:
Oh I am a silly goose! I got this now. I fixed up the plots quite a bit. They should be much better.